tools/azure-sdk-tools/devtools_testutils/proxy_testcase.py

```diff
@@ -109,25 +116,26 @@ def start_record_or_playback(test_id: str) -> "Tuple[str, Dict[str, str]]":
 
 
 def stop_record_or_playback(test_id: str, recording_id: str, test_variables: "Dict[str, str]") -> None:
-    if is_live():
-        response = requests.post(
-            RECORDING_STOP_URL,
-            headers={
-                "x-recording-file": test_id,
-                "x-recording-id": recording_id,
-                "x-recording-save": "true",
-                "Content-Type": "application/json",
-            },
-            json=test_variables or {},  # tests don't record successfully unless test_variables is a dictionary
-        )
-    else:
-        response = requests.post(
-            PLAYBACK_STOP_URL,
-            headers={"x-recording-id": recording_id},
-        )
     try:
-        response.raise_for_status()
-    except requests.HTTPError as e:
+        if is_live():
+            http_client.request(
+                method="POST",
+                url=RECORDING_STOP_URL,
+                headers={
+                    "x-recording-file": test_id,
+                    "x-recording-id": recording_id,
+                    "x-recording-save": "true",
+                    "Content-Type": "application/json",
+                },
+                json=test_variables or {},  # tests don't record successfully unless test_variables is a dictionary
```

Comment:
use `body` instead of `json`

Reply:
@pvaneck thank you so much for this feedback, I'm going to work off it all and get this thing updated to work properly.